### PR TITLE
Makes MobileNetV3 a Functional subclass model & removes closures

### LINKED ISSUE #1427: Make MobileNetV3 a Functional subclass model & remove closures
This should follow the example of #1412 and #1401, including

- Docstring updates
- Colab demonstrating that weight loading works (if pre-trained weights exist) and numerics are identical

### PR DESCRIPTION
**Pull request by ID6109**
# What does this PR do?

Fixes #1427 

[Colab](https://colab.research.google.com/drive/1ToRLkxVjLzkBVBuIjlOmIdNaiNCbkCO8?usp=sharing)

## Who can review?
@ianstenbit @LukeWood @tanzhenyu

### CODE REVIEW

File: keras_cv/models/mobilenet_v3.py [hunk 1]
```diff
@@ -112,48 +112,44 @@ def depth(x, divisor=8, min_value=None):
     return new_x
 
 
-def HardSigmoid(name=None):
+def HardSigmoid(x, name=None):
```
Comment (ianstenbit) [inline keras_cv/models/mobilenet_v3.py:115]: let's rename these activation functions (and the InvertedResBlock) to use the `apply_xyz` convention

(e.g. `apply_hard_sigmoid`, `apply_inverted_res_block`)

File: keras_cv/models/mobilenet_v3.py [hunk 2]
```diff
@@ -179,104 +176,86 @@ def InvertedResBlock(
...
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class MobileNetV3(keras.Model):
     """Instantiates the MobileNetV3 architecture.
 
     References:
         - [Searching for MobileNetV3](https://arxiv.org/pdf/1905.02244.pdf) (ICCV 2019)
         - [Based on the Original keras.applications MobileNetv3](https://github.com/keras-team/keras/blob/master/keras/applications/mobilenet_v3.py)
 
-    This function returns a Keras MobileNetV3 model.
+    This class returns a Keras MobileNetV3 model.
```
Comment (ianstenbit) [inline keras_cv/models/mobilenet_v3.py:258]: While you're editing this, would you mind also adding type indications to these docstrings?

(See, for example, docstring updates to VGG19 in #1471)

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thank you for the PR!
Just a few small changes, otherwise looks good!

**Comment by ID6109** [comment]
The changes have been made. I'll implement the same in my other PRs. Do let me know if there are further corrections needed. Thanks.

File: keras_cv/models/mobilenet_v3.py [hunk 3]
```diff
@@ -72,33 +72,33 @@
...
+        name: string, optional name to pass to the model, defaults to "{name}".
 
     Returns:
         A `keras.Model` instance.
 """
 
 
-def depth(x, divisor=8, min_value=None):
-    """Ensure that all layers have a channel number that is divisble by the `divisor`.
+def apply_depth(x, divisor=8, min_value=None):
```
Comment (ianstenbit) [inline keras_cv/models/mobilenet_v3.py:91]: nit: this isn't a function on a Tensor, but a helper method to get the number of channels for a convolutional layer, so it should just be `depth` instead of `apply_depth`

File: keras_cv/models/mobilenet_v3.py [hunk 4]
```diff
@@ -112,48 +112,44 @@ def depth(x, divisor=8, min_value=None):
...
+def apply_hard_sigmoid(x, name=None):
     """The Hard Sigmoid function.
 
     Args:
+        x: input tensor
         name: string, layer label.
 
     Returns:
-        a function that takes an input Tensor representing a HardSigmoid layer.
+        the updated input tensor.
```
Comment (ianstenbit) [inline keras_cv/models/mobilenet_v3.py:123]: (throughout) - it would be more accurate to say that this returns "the result of applying hard sigmoid to the input tensor"

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks for the changes!

Just a few more small changes, and then we can merge.

File: keras_cv/models/mobilenet_v3.py (hunk 4 above)
Comment (ianstenbit) [reply keras_cv/models/mobilenet_v3.py:123]: Actually, this looks consistent with the docs in other models, so feel free to ignore this for now and we can address it in all the models at once in #1500

Comment (ID6109) [reply keras_cv/models/mobilenet_v3.py:123]: I'll take this up directly while solving #1500 then.

File: keras_cv/models/mobilenet_v3.py [hunk 5]
```diff
@@ -72,33 +72,33 @@
...
 
-def depth(x, divisor=8, min_value=None):
-    """Ensure that all layers have a channel number that is divisble by the `divisor`.
+def apply_depth(x, divisor=8, min_value=None):
+    """Ensure that all layers have a channel number that is divisible by the `divisor`.
 
     Args:
-        x: input value.
-        divisor: integer, the value by which a channel number should be divisble,
+        x: input tensor.
```
[the remark this answers is not in the document]
Comment (ID6109) [reply keras_cv/models/mobilenet_v3.py:95]: It seems I skimmed past this method in a hurry. I'll be more mindful in the future. Thanks.

[the remark this answers is not in the document]
Comment (ianstenbit) [reply keras_cv/models/mobilenet_v3.py]: No worries at all -- thank you for the contributions!

**Review by ianstenbit — APPROVED** [review]
Looks good -- thank you!
